### PR TITLE
Port hostmaint alarms function to Knative

### LINKED ISSUE #856: Port hostmaint-alarm function from OpenFaaS to Knative
**Describe the solution you'd like**
Port the PowerCLI disable hostmaint alarm function from OpenFaaS to Knative

### PR DESCRIPTION
**Pull request by kremerpatrick**
Closes #856

Signed-off-by: Patrick Kremer <pkremer@vmware.com>

## Summary

Porting the host maintenance alarms function from OpenFaaS to Knative. 

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

List of Issues closed or resolved by this PR. Add multiple `Closes` keyword followed by the issue number (e.g. Closes #ISSUE-NUMBER)

Closes #856

## Testing Verification

Tested on VEBA 0.7.0

### CODE REVIEW

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+```console
+docker push ${IMAGE}
+```
+
+Update the `hostmaint_secret.json` file with your vCenter Server credentials and configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `HOSTMAINT_SECRET`.
+
+```console
+# create secret
+
```
Comment (embano1) [inline examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:144]: nit: newline

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+kubectl -n vmware-functions create secret generic hostmaint-secret --from-file=HOSTMAINT_SECRET=hostmaint_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret hostmaint-secret app=veba-ui
+```
+
+Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `DvsReconfiguredEvent` vCenter Server Event. If you wish to change this, update the `subject` field within `function.yaml` to the desired event type.
```
Comment (embano1) [inline examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:151]: Is the event name correct?

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+
+Deploy the function to the VMware Event Broker Appliance (VEBA).
+
+```console
+# deploy function
+
```
Comment (embano1) [inline examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:158]: nit: newline

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md (hunk 1 above)
Comment (kremerpatrick) [reply examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:144]: fixed

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+kubectl -n vmware-functions create secret generic hostmaint-secret --from-file=HOSTMAINT_SECRET=hostmaint_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret hostmaint-secret app=veba-ui
+```
+
+Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `DvsReconfiguredEvent` vCenter Server Event. If you wish to change this, update the `subject` field within `function.yaml` to the desired event type.
+
+
```
[the remark this answers is not in the document]
Comment (kremerpatrick) [reply examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:153]: fixed

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md (hunk 3 above)
Comment (kremerpatrick) [reply examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:158]: Fixed

File: examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+annotations:
+  autoscaling.knative.dev/maxScale: "1"
+  autoscaling.knative.dev/minScale: "1"
+```
+
+# Step 4 - Undeploy
+
+```console
+# undeploy function
+
```
[the remark this answers is not in the document]
Comment (kremerpatrick) [reply examples/knative/powercli/kn-pcli-hostmaint-alarms/README.md:174]: fixed

**Review by embano1 — APPROVED** [review]
LGTM!

**Pull request merged**